### PR TITLE
fix(api): resolve pool-session path-aliases in session target resolver

### LINKED ISSUE #1873: bug: API session-resolver doesn't honor pool-session path-aliases (affects /extmsg/inbound, gc session nudge, any caller-by-name)
## Summary

`POST /v0/city/<name>/extmsg/inbound` silently drops messages when a binding's `participants[].session_name` is a path-alias / pool-session name (not a `[[named_session]]` entry in city.toml). The dispatch resolver only handles raw session IDs, exact-ID-match named-sessions, and `apiTemplateTarget` patterns — pool-session path-aliases (the kind `gc session list` shows under `TARGET` / `TITLE`) fall through to "session not found" and the inbound is dropped.

## Symptom

Operator binds a slack room to a project-lead via `gc slack bind-room`, using the path-alias string they see in `gc session list`. Subsequent inbound messages from the bound room arrive at the gascity API, get logged as 200 OK, but never reach the bound session — supervisor logs:

```
2026/05/08 22:10:41 api: POST /v0/city/ds-research/extmsg/inbound 200 582.832ms [memory]
2026/05/08 22:10:41 extmsg: resolve session gascity-maintenance-pl failed: session not found: "gascity-maintenance-pl"
```

The agent's session is alive (`gc session list` shows it `active`); only the resolver doesn't recognize its identifier.

## Reproduction

1. Stand up a city with at least one rig that has a pool-style project-lead agent (ours is `[[agent]] name = "gascity-maintenance-pl"` with a path template, not a named-session entry).
2. Confirm the session shows up in `gc session list` with both its ID (e.g. `gc-149957`) and a stable path-alias / TITLE (e.g. `gascity-maintenance-pl`).
3. Bind a slack room to the path-alias:
   ```
   gc slack bind-room <CHANNEL_ID> gascity-maintenance-pl <other-participant> \
     --default-handle gascity-maintenance-pl \
     --handle gascity-maintenance-pl=gascity-maintenance-pl \
     --handle <other>=<other>
   ```
   Bind succeeds; `.gc/services/slack/data/config.json` shows `participants[0].session_name == "gascity-maintenance-pl"`.
4. Post a non-targeted message into the slack room.
5. Adapter logs `inbound: chan=… target=""`; gascity API logs `extmsg: resolve session gascity-maintenance-pl failed: session not found`. The session never sees the inbound. No emoji, no reply.

Workaround: re-bind with the **raw session ID** (`gc-149957`) instead of the path-alias. Resolves correctly. But raw IDs go stale on every supervisor restart, so the operator has to re-bind after every cutover — exactly the toil the path-alias was supposed to absorb.

Result: `inbound delivered to bound session` → got `inbound silently dropped, dispatch logs say "session not found"`.

## Root cause

`internal/api/session_resolution.go:364-407` (`resolveSessionTargetIDWithContext`) tries, in order:

1. `parseAPITemplateTarget(identifier)` — template/path patterns (line 368).
2. `session.ResolveSessionIDByExactID` — raw ID match (line 371).
3. `resolveConfiguredNamedSessionIDWithContext` — `[[named_session]]` config entries only (line 376).
4. `session.ResolveSessionID` — beads-store name lookup (line 381).
5. (Optionally) `ResolveSessionIDAllowClosed` (line 400).

A path-alias for a pool session (not a `[[named_session]]`) matches none of these. Step 3 is gated on a city-config named-session entry; Step 4's `ResolveSessionID` looks for an exact bead-store name match, which doesn't include the path-alias used by `gc session list`. So the identifier falls to `apiSessionTargetNotFound` (line 406), and `internal/api/handler_extmsg.go:154-157` logs the failure and skips delivery.

The asymmetry: `bind-room` accepts and stores any string as `participants[].session_name` without validation. The resolver only honors a strict subset. Operators cannot tell ahead of time which strings will resolve at dispatch time.

## Fix candidates

**A. Resolver-side: extend `resolveSessionTargetIDWithContext` to recognize pool-session path-aliases.**
- Add a step that walks `gc session list`-style live sessions and matches `identifier` against their TITLE / path-alias.
- Scope: ~30 LOC in `internal/api/session_resolution.go`, plus a unit test covering pool-session path-aliases.
- Trade-off: makes the resolver more permissive — same identifier could theoretically match multiple live sessions (though path-aliases are unique per agent in practice). Would need a tiebreaker rule.

**B. Bind-side: validate at `gc slack bind-room` (and other binding entry points) that every `session_name` resolves at bind time, fail loud if not.**
- Scope: ~20 LOC in the slack-pack bind script, plus equivalent validation on `POST /extmsg/participants`.
- Trade-off: shifts the failure earlier (bind fails instead of silent drop) but doesn't restore the convenience of path-aliases — operator still has to use raw IDs. Worse ergonomics; better diagnostics.

**Recommendation: A.** The whole point of path-aliases is that they're stable across supervisor restarts and pool re-spawns; raw session IDs are not. The resolver should honor what `bind-room` writes; otherwise bindings have to be repaired manually after every restart, which is exactly the toil the previous mayor session described in `feedback_slack_adapter_race.md`. B is worth doing as a defensive layer regardless, but A is the substantive fix.

## Adjacent / out of scope

- `gc slack bind-room`'s default-handle behavior (no dispatch when `target=""` and the binding's `default_handle` doesn't resolve) is downstream of this — once the resolver works for path-aliases, default-handle dispatch starts working too.
- The eyes-react / threading protocol issues observed in the same session are slack-pack concerns (per-provider agent protocol owned by the provider-pack), not gascity. Tracking separately.
- PR #1869's parameterization of the inbound reminder text on `conv.Provider` is orthogonal — that's about reminder content; this is about reminder *delivery*.

## Maintainer context

Surfaced during slack-pack cutover dogfooding on `ds-research` (mayor's city). After supervisor restart drained the maintenance-pl pool and re-spawned with a new session ID (`gc-149448` → `gc-149957`), the channel binding was re-issued with the path-alias `gascity-maintenance-pl` as participant `session_name` — expecting the path-alias to be the durable identifier. First two operator pings into `#gascity-maintenance` were silently dropped; rebinding with the raw `gc-149957` ID restored delivery. Reproduces 100% on origin/main `f9ef3d68`.

### PR DESCRIPTION
**Pull request by sjarmak**
Closes #1873.

## Summary

`/extmsg/inbound` dispatch and `gc session nudge` previously returned "session not found" for any identifier that matched a live pool-session's `Title` (path-alias) but not a `[[named_session]]` config entry, an exact session ID, or a recognized template target. Pool sessions surface their stable path-alias under `Title` (the same string `gc session list` shows under TARGET / TITLE) while their `session_name` is a synthetic internal id (`s-gc-NNN`), so they were invisible to the existing resolver chain.

This adds a fall-through resolver step (`resolveLiveSessionByPathAlias`) that matches the identifier against active pool-session Titles. It runs after `session.ResolveSessionID` so explicit session_name/alias matches always win.

## Repro on origin/main

```
$ gc session list --json | jq '.[] | select(.Title == "gascity-maintenance-pl" and .State == "active") | .ID'
"gc-149957"
$ gc session nudge gascity-maintenance-pl "test"
gc session nudge: session not found: "gascity-maintenance-pl"
```

The session is `active`, has a stable Title that matches the lookup, and the resolver still bails. Same shape as `/extmsg/inbound` dispatch failures observed when slack bind-room participants are written using path-aliases (the durable identifier that survives supervisor restarts) instead of raw session IDs (which go stale).

## Behavior after fix

```
$ gc session nudge gascity-maintenance-pl "test"
# resolves to gc-149957, message delivered
```

## Resolver chain order (final)

1. `parseAPITemplateTarget` — template-format identifiers
2. `session.ResolveSessionIDByExactID` — exact `gc-NNN` ID match
3. `resolveConfiguredNamedSessionIDWithContext` — `[[named_session]]` config entries (with orphan-rejection)
4. `session.ResolveSessionID` — `session_name` / `alias` index lookup (with orphan-rejection)
5. **`resolveLiveSessionByPathAlias` (new)** — `Title` match for active pool sessions
6. `opts.allowClosed` fallback

The new step runs **after** `session.ResolveSessionID` so explicit session_name/alias matches always win — preventing a cross-bead collision where a pool session's Title shadows a different bead's session_name.

## State filter rationale

Accepts only `{active, awake}`. Excluded states intentionally fall through to `apiSessionTargetNotFound`:

- **asleep** — not running, can't receive messages
- **draining** — on its way out, shouldn't get new external messages
- **creating** — runtime still booting; `sendBackgroundMessageToSession` would deliver against an incomplete provider, worse than not-found. Once the reconciler flips state=active, subsequent inbounds resolve cleanly.

`StateAwake` is included because the lifecycle projection (`internal/session/lifecycle_projection.go`) returns `StateAwake` whenever runtime is observed alive — pool sessions reach this state, not just named-sessions.

## Tiebreaker

When multiple active pool sessions share the same Title (rare misconfiguration), the most-recently-created bead wins. Ties on `CreatedAt` resolve to the first match in store iteration order. The doc comment notes this explicitly.

## Test coverage

15 tests in `internal/api/session_resolution_path_alias_test.go`:

- Happy-path: single active pool session resolves by Title (state `active` and state `awake`)
- Tiebreaker: two active pool sessions with same Title — most-recent wins; verified order-independent via reversed input
- State filter: `closed`, `asleep`, `draining`, `creating` all return `ErrSessionNotFound`
- Named-session skip: configured named-session bead with matching Title is not accidentally caught
- Exact-ID precedence: raw `gc-NNN` match still wins over a different bead's Title-match
- Configured-named precedence: `[[named_session]]` config entry still wins over Title-match on a different bead
- **Cross-bead collision**: when one bead's `session_name` matches and a different bead's `Title` matches the same identifier, `session.ResolveSessionID` (step 4) wins over the new path-alias step (step 5). Guards the resolver-ordering invariant
- Empty-store and empty-identifier edge cases
- Nil-store edge case
- End-to-end via `resolveSessionTargetIDWithContext` (the entry point used by `/extmsg/inbound`)

The tiebreaker test uses an in-test fake store with explicit `CreatedAt` values rather than `time.Sleep`, so it's deterministic regardless of host clock resolution or load.

## Mechanical gates

- `make build` ✅
- `make check` (fmt-check, lint, vet, full test suite) ✅
- `make check-docs` not run — no docs/engdocs touched

## Commits

- `8bc2001d` fix(api): resolve pool-session path-aliases in session target resolver
- `d7b4d796` test+doc: address review findings — StateCreating doc + deterministic tiebreaker test + StateDraining/StateCreating skip coverage
- `a711c7bb` fix-up: address /gascity-ship Stage 2 review findings (resolver reorder + doc trim + redundant `Status == "closed"` check removal + cross-bead-collision test)

## Multi-model review

Pre-push pipeline ran simplify + multi-model review (Anthropic ×3 lenses + Codex grpt-5.4) + 29-rule contributor check. All blockers and majors addressed in the fix-up commits; remaining minors (CachingStore staleness, repeated test-helper boilerplate, equal-`CreatedAt` tiebreaker fallback) are documented as accepted trade-offs.

Codex specifically caught the cross-bead resolver-ordering concern that prompted putting `resolveLiveSessionByPathAlias` after `session.ResolveSessionID` — single-pass review would have shipped the original ordering.

---
<a href="https://app.blacksmith.sh/gastownhall/codesmith/gascity/pr/1897"><picture><source media="(prefers-color-scheme: dark)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-in-codesmith-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://pr-comments-assets.blacksmith.sh/codesmith/view-in-codesmith-light.svg">View in Codesmith</picture></a>
<sup>Need help on this PR? Tag <code>@codesmith</code> with what you need.</sup>

- [ ] Let Codesmith autofix CI failures and bot reviews

### CODE REVIEW

File: internal/api/session_resolution.go [hunk 1]
```diff
@@ -361,6 +361,69 @@ func (s *Server) materializeNamedSession(store beads.Store, spec apiNamedSession
...
+		if !session.IsSessionBeadOrRepairable(b) {
+			continue
+		}
+		if apiIsNamedSessionBead(b) {
+			continue
+		}
+		if strings.TrimSpace(b.Title) != identifier {
+			continue
+		}
+		state := session.State(b.Metadata["state"])
```
[the remark this answers is not in the document]
Comment (sjarmak) [reply internal/api/session_resolution.go:415]: Good catch — fixed in 6eb19cce. Verified the codebase convention: `internal/session/manager.go:741,813` explicitly normalize `current == StateNone` to `StateActive` in the reconciler paths. Updated the path-alias resolver's state filter to match (`{active, awake, none}`) and added `TestResolveSessionTargetID_PathAliasEmptyStateTreatedAsActive` to cover the legacy-bead case. Doc comment now references the manager.go normalization for the next reader.

File: internal/api/session_resolution_path_alias_test.go [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+	_, err := srv.resolveSessionIDWithConfig(fs.cityBeadStore, "asleep-pl")
+	if !errors.Is(err, session.ErrSessionNotFound) {
+		t.Fatalf("resolveSessionIDWithConfig(asleep path-alias) = %v, want ErrSessionNotFound", err)
+	}
+}
+
+// TestResolveSessionTargetID_ExactIDWinsOverPathAlias seeds two beads where
+// one is addressable by exact ID and another shares its ID string as a
+// path-alias on a different (active) session. The exact-ID branch (step 2)
+// must win before the path-alias branch (step 4) runs.
```
[the remark this answers is not in the document]
Comment (sjarmak) [reply internal/api/session_resolution_path_alias_test.go]: Fixed in 6eb19cce — updated the test comment to reference "step 5" matching the actual resolver order (path-alias step now runs after `session.ResolveSessionID` per the resolver-chain reorder in commit a711c7bb). The behavior assertion is unchanged (exact-ID still wins); only the step number in the comment was stale.

File: internal/api/session_resolution_path_alias_test.go [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+	}
+	if id != target.ID {
+		t.Fatalf("resolved id = %q, want exact-ID bead %q", id, target.ID)
+	}
+}
+
+// TestResolveSessionTargetID_ConfiguredNamedSessionWinsOverPathAlias seeds
+// a configured named-session with identity "myrig/worker" alongside a pool
+// session whose Title shadows that identity. The configured-named-session
+// branch (step 3) must win before the path-alias branch (step 4).
```
[the remark this answers is not in the document]
Comment (sjarmak) [reply internal/api/session_resolution_path_alias_test.go]: Fixed in 6eb19cce — same step-number drift as the prior comment. Updated to reference "step 5" so future maintainers reading the test comment see the current resolver order. Behavior unchanged: configured named-session still wins over a Title-shadow on a different bead.
